packages/renderer/src/lib/actions/ContributionActions.svelte

```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+import type { Menu } from '../../../../main/src/plugin/menu-registry';
+import { faEllipsisVertical } from '@fortawesome/free-solid-svg-icons';
+import ListItemButtonIcon from '../ui/ListItemButtonIcon.svelte';
+
+function isSerializable(value: any): boolean {
+  switch (typeof value) {
+    case 'string':
+    case 'number':
+    case 'boolean':
+    case 'object':
+      return true;
+    default:
+      return false;
+  }
+}
+
+function removeNonSerializableProperties<T>(obj: T): T {
+  if (typeof obj !== 'object' || obj === undefined) {
+    return obj;
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map(item => removeNonSerializableProperties(item)) as any;
+  }
+
+  const result: Partial<T> = {};
+
+  for (const key in obj) {
+    if (isSerializable(obj[key])) {
+      result[key] = removeNonSerializableProperties(obj[key]);
+    }
+  }
+
+  return result as T;
+}
```

Comment:
you could extract that to a .ts util file so we can have simple unit tests not involving testing-library and it can be shared across multiple components